openverse_catalog/dags/providers/provider_api_scripts/wikimedia_commons.py

```diff
@@ -23,94 +112,161 @@
+        self.popularity_cache: dict[str, int] = {}
 
     def get_next_query_params(self, prev_query_params, **kwargs):
+        # NOTE: If more sub-properties are added here, an additional contingency for
+        # NOTE: iterating over that property's "continue" token will need to be added
+        # NOTE: to the `adjust_parameters_for_next_iteration` function, otherwise the
+        # NOTE: ingestion could get stuck in an infinite loop.
         return {
+            # The type of action being taken for the API request
             "action": "query",
+            # What "generator" to use for returning results, in this case "allimages"
             "generator": "allimages",
+            # How to sort the "global all images" (gai) results
             "gaisort": "timestamp",
+            # How to order the results
             "gaidir": "newer",
+            # The number of "all images" results to return per request
             "gailimit": self.batch_limit,
-            "prop": "imageinfo|globalusage",
-            "iiprop": "url|user|dimensions|extmetadata|mediatype|size|metadata",
+            # The number of "global usage" results to return per request
             "gulimit": self.batch_limit,
+            # What namespace to search in, 0 is the main namespace
+            # https://www.mediawiki.org/wiki/Help:Namespaces
             "gunamespace": 0,
             "format": "json",
+            # The timestamp to start the search from
             "gaistart": self.start_timestamp,
+            # The timestamp to end the search at
             "gaiend": self.end_timestamp,
+            # The current properties to use for the request
+            **self.current_props,
+            # The current continue token to use for the request provided by Wikimedia
             **self.continue_token,
         }
 
     def get_media_type(self, record):
         """Get the media_type of a parsed Record"""
         return record["media_type"]
 
-    def get_response_json(self, query_params):
+    def get_response_json(self, query_params, **kwargs):
         """
         Get the response data from the API.
 
         Overrides the parent function to make multiple requests until
         we see "batchcomplete", rather than a single request to the
         endpoint. This ensures that global usage data used for calculating
         popularity is tabulated correctly.
+
+        This also dynamically adjust the query parameters based on the response sizes
+        for a given continue token. For more information see the DAG description.
         """
         batch_json = None
+        gaicontinue = None
+        iteration_count = 0
 
-        for _ in range(MEAN_GLOBAL_USAGE_LIMIT):
+        for _ in range(self.mean_global_usage_limit):
             response_json = super().get_response_json(
                 query_params,
                 timeout=60,
             )
 
             if response_json is None:
+                logger.warning("Response JSON is None")
```

Comment:
Oh shoot, that's a really good point. I don't know when this would occur, but it was a case that was handled previously 🤷🏼‍♀️ Maybe we should raise an exception instead so we can see cases where this might occur?

Honestly looking at this more, I don't think this value should _ever_ be `None` when returned from the DelayedRequester, so I think I'll just remove the condition! Perhaps it was vestigial code from before we unified the requester interface 🙂